apisix/core/response.lua

```diff
@@ -80,7 +115,20 @@ function resp_exit(code, ...)
     end
 
     if idx > 0 then
-        ngx_print(concat_tab(t, "", 1, idx))
+        local response = concat_tab(t, "", 1, idx)
+        local _, err = decode_json(response)
```

Comment:
It would be more effective to check the header first before decoding, so we can save lots of unnecessary decoding.

Reply:
Fiexed by @MekelCon
[perf: decode json later](https://github.com/apache/apisix/pull/8587/commits/b4b60d28eb1ec0b1c0078633219d846e7be839a6)